net-wireless/wireless-regdb/wireless-regdb-20230503.ebuild

```diff
@@ -13,25 +15,47 @@ LICENSE="ISC"
 SLOT="0"
 KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~loong ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86"
 
+REQUIRED_USE="kernel_linux"
+
+pkg_pretend() {
+	if kernel_is -ge 4 15 && has_version net-wireless/crda; then
+		ewarn "Starting from kernel version 4.15 net-wireless/crda is no longer needed."
+		ewarn "It can be safely uninstalled."
+	fi
+
+	CONFIG_CHECK="~CFG80211_CRDA_SUPPORT"
+	WARNING_CFG80211_CRDA_SUPPORT="REGULATORY DOMAIN PROBLEM: \
+please enable CFG80211_CRDA_SUPPORT for proper regulatory domain support"
+
+	check_extra_config
+
+	if linux_config_exists && linux_chkconfig_builtin CFG80211 &&
+		[[ $(linux_chkconfig_string EXTRA_FIRMWARE) != *regulatory.db* ]]
+	then
+		ewarn "REGULATORY DOMAIN PROBLEM:"
+		ewarn "  With CONFIG_CFG80211=y (built-in), the driver won't be able to load regulatory.db from"
+		ewarn "  /lib/firmware, resulting in broken regulatory domain support. Please set CONFIG_CFG80211=m"
+		ewarn "  or add regulatory.db and regulatory.db.p7s to CONFIG_EXTRA_FIRMWARE."
+	fi
+}
+
 src_compile() {
-	einfo "Recompiling regulatory.bin from db.txt would break CRDA verify. Installing unmodified binary version."
+	true
 }
 
 src_install() {
+	if kernel_is -lt 4 15; then
```

Comment:
this is flat wrong.  The kernel checks above are not fatal (nor should they be) so users may be on a newer kernel which isn't configured properly and need crda.  I've said multiple times to add a crda use flag, I know you hate crda, but a non-default use flag is going to provide the best user experience here.